### DISCUSSION TITLE
When using standalone: true created node_modules does not include packages from server.js

### DISCUSSION THREAD

**Discussion by samueldusek** [vercel/next.js#35961] [Help]
I am using the new experimental build setting: [`standalone: true`](https://nextjs.org/blog/next-12-1?utm_source=next-site&utm_medium=banner&utm_campaign=next-website#self-hosted-nextjs-improvements) in my `./next.config.js` file. But for some reason the packages that are being required in my custom `./server.js` file are not being included in the `./next/standalone/node_modules` folder. Is there any reason for that?

How can I use the `standalone` setting with my custom server? 

Thanks a lot! 🙂

**Comment by sladg** [comment]
I'm facing the same issue currently! Works fine for a smaller project, for bigger with yarn (using yarn@3) it does not.

**Comment by thomasdvd** [comment]
This happened to me while using sequelize with optional peer dependency tedious. So as a hack i simply added an import to a random file in the app and used the variable, import without using the variable is not enough

```typescript
import * as td from "tedious";
console.log(td)
```

after that the package "tedious" was included in `./next/standalone/node_modules`

**Comment by larsqa** [comment]
Can you share an example how you use a custom server? Related to [#34599](https://github.com/vercel/next.js/discussions/34599)

**Comment by sibstark** [comment]
Faced the same issue - why do we need node_modules inside standalone?

**Comment by crice88** [comment]
Not sure if you're using docker, but I had to manually install the packages that I needed that weren't included in the standalone bundle. It would be nice if you could flag some packages as dependent.

**Comment by christopherpickering** [comment]
for anyone else trailing in here, here is some docs for this now: https://nextjs.org/docs/app/api-reference/config/next-config-js/output#caveats

I needed prisma and node-fetch to stay:
```ts
const nextConfig = {
  reactStrictMode: true,
  output: 'standalone',
  outputFileTracingIncludes: {
    '/api/': ['./node_modules/@prisma/**/*', './node_modules/node-fetch/**/*'],
  },
}
```

but how to get packages that are imported in those 🥹
